flush-in-loop-behavior.markdown

```diff
@@ -0,0 +1,48 @@
+
+
+Take this example:
+
+```typescript
+const authors = await em.find(Author, { id: ["1", "2"] } );
+await Promise.all(
+  authors.map(async (a) => {
+    const p = await em.findOneOrFail(Publisher, { id: a.publisher.id });
+    a.firstName = a.firstName + p.name;
+    await em.flush();
```

Comment:
Granted, this example is somewhat contrived b/c the `em.flush()` could just be moved outside the loop, but often times you'll have business logic that is still inside the loop, but after the `em.flush`, i.e. you needed to flush to get some ids assigned or something, and then want both lambdas to continue doing work.